### DISCUSSION TITLE
how to customize style of inline code blocks only

### DISCUSSION THREAD

**Discussion by hopezh** [squidfunk/mkdocs-material#7042] [Q&A] [answered]
# what I want to do
I'd like to customize bg and fg color of inline code blocks, and change them from image 1 shown below to image 2. 

# what I tried
I tried the following, according to this [guide](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/?h=code#custom-syntax-theme), which changes the style of all code blocks, including normal code blocks: 

```css
:root>* {
    --md-code-fg-color: #ffffff;
    --md-code-bg-color: #0000ff;
}
```

# images

image 1:
![2024-04-11 23 16 25](https://github.com/squidfunk/mkdocs-material/assets/23435810/cf40dc6f-caa9-477d-a560-d99aa6519c89)

image 2:
![2024-04-11 23 16 47](https://github.com/squidfunk/mkdocs-material/assets/23435810/d2ccf61d-33a7-4f63-8e33-aa1527f3cd31)

**Comment by aborcyjnydreamteam** [comment, marked as answer]
Something like this?

```css
.md-typeset > p > code {
    background-color: red;
}
```

**Comment by hopezh** [comment]
hi @aborcyjnydreamteam, may I ask you a follow up question? 

Your suggestion works for inline code block in **body text**, but not for those in **bullet point text** (which is actually what is shown in the image I posted above.). So, is there a CSS setting for this specifically? 

Thanks.

**Reply by facelessuser** [reply]
Well, if you only specify that code _must_ be a direct child of `p`, then it won't work in lists and other places. It would be better to simply distinguish if from block code.

Block code is wrapped in pre tags: `<pre><code></code></pre>`. Inlline code is not. So maybe this is more helpful.

```css
.md-typeset *:not(pre) > code {
  background-color: red;
}
```

**Reply by hopezh** [reply]
Thanks, @facelessuser 

You suggestion works like a charm. 

However, the background color of normal code block is also affected, as shown below. 

I think I'll use the default bg color for inline code within bullet points, for the time being. 

# before
![2024-04-14 12 21 17](https://github.com/squidfunk/mkdocs-material/assets/23435810/c3f88007-6579-43ed-944f-48770fe92d42)

# after
![2024-04-14 12 21 39](https://github.com/squidfunk/mkdocs-material/assets/23435810/55de4df1-80c2-447a-a464-7507d792125d)

**Reply by facelessuser** [reply]
I was probably making some assumptions about format. With my personal configuration, it works fine. I think when using line numbers, the default uses a table format... I'd have to refresh myself on the exact HTML structure used and update the rule. The basic idea of what I'm suggesting should still work, but we need to target the exact, specific differences. I'll take a look tomorrow as it's late here and come back with a response.

**Reply by hopezh** [reply]
Thank you very much for being so patient with such a small inquiry.

**Reply by facelessuser** [reply]
I added this CSS directly to the official MkDocs Material site (in the browser):

```css
.md-typeset *:not(pre) > code {
    background: blue;
    color: white;
}
```

It worked as I described:

<img width="796" alt="Screenshot 2024-04-15 at 8 54 39 AM" src="https://github.com/squidfunk/mkdocs-material/assets/1055125/cfa602da-1fa0-4753-9382-b618a21b7ce1">

If you are doing custom stuff, maybe it won't work the same, but based on the general, recommended configuration, the suggested CSS should work as demonstrated in the image, for both code with line numbers and without.

**Reply by hopezh** [reply]
@facelessuser Thank you. Yes, your suggestion works like a charm, again. 

The key is the `>` symbol in your `CSS` snippet, which I didn't add previously. 

So, the following is what I have now, after this `CSS` code in `docs/stylesheets/extra.css` in my `mkdocs` project folder. 

`inline code block` in either body text or bullet points are customized together now, without affecting the background color of normal code blocks. 

![2024-04-18 15 49 53](https://github.com/squidfunk/mkdocs-material/assets/23435810/f0708706-4a2f-4a0f-a5f2-df6a999f4131)
